test/integration/master/kube_apiserver_test.go

```diff
@@ -138,3 +141,97 @@ func TestOpenAPIDelegationChainPlumbing(t *testing.T) {
 		t.Errorf("missing path: %q", registrationPrefix)
 	}
 }
+
+func getEndpointIPs(endpoint corev1.Endpoints) []string {
+	endpoints := make(map[string]bool)
+	ips := make([]string, 0)
+	for _, subset := range endpoint.Subsets {
+		for _, address := range subset.Addresses {
+			if _, ok := endpoints[address.IP]; !ok {
+				endpoints[address.IP] = true
+				ips = append(ips, address.IP)
+			}
+		}
+	}
+	return ips
+}
+
+func TestReconcilerMasterLeaseCombined(t *testing.T) {
+	var servers []*kubeapiservertesting.TestServer
+	etcd := framework.SharedEtcd()
+
+	// start lease api servers
+	leaseCount := 2
+	for i := 0; i < leaseCount; i++ {
+		options := []string{
+			"--endpoint-reconciler-type", "lease",
+			"--advertise-address", fmt.Sprintf("10.0.1.%v", i+10),
+		}
+		server := kubeapiservertesting.StartTestServerOrDie(t, options, etcd)
+		defer server.TearDownFn()
+		servers = append(servers, server)
+	}
+
+	// start master count api server
+	masterCountServer := kubeapiservertesting.StartTestServerOrDie(t, []string{
+		"--endpoint-reconciler-type", "master-count",
+		"--advertise-address", "10.0.1.1",
+	}, etcd)
+
+	// wait until we eventually fetch the master count reconciler. the lease and
+	// master count API server endpoints will flap back and forth
+	if err := wait.PollImmediate(3*time.Second, 1*time.Minute, func() (bool, error) {
+		client, err := kubernetes.NewForConfig(servers[0].ClientConfig)
+		if err != nil {
+			t.Logf("create client error: %v", err)
+			return false, nil
+		}
+		endpoints, err := client.CoreV1().Endpoints("default").List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("error fetching endpoints: %v", err)
+			return false, nil
+		}
+		ips := getEndpointIPs(endpoints.Items[0])
+		return len(ips) == 1, nil
+	}); err != nil {
+		t.Fatalf("did not find master-count api server: %v", err)
+	}
+
+	if err := wait.PollImmediate(3*time.Second, 1*time.Minute, func() (bool, error) {
+		client, err := kubernetes.NewForConfig(servers[0].ClientConfig)
+		if err != nil {
+			t.Logf("create client error: %v", err)
+			return false, nil
+		}
+		endpoints, err := client.CoreV1().Endpoints("default").List(metav1.ListOptions{})
```

Comment:
Go through the "kubernetes" service explicitly? Just to make sure we're not conflating this with other endpoints in the default namespace.

Reply:
yes, get the endpoints object named `kubernetes`

Reply:
fixed... added some comments throughout as well.